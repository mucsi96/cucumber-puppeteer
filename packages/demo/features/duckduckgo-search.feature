Feature: DuckDuckGo

Scenario: Searching DuckDuckGo

  Given I open DuckDuckGo search page
  Then the DuckDuckGo search page is displayed
  And the title is "DuckDuckGo — Privacy, simplified."

Scenario: Searching DuckDuckGo again

  Given I open DuckDuckGo search page
  Then the title is "DuckDuckGos — Privacy, simplified."
  And the DuckDuckGo search form exists