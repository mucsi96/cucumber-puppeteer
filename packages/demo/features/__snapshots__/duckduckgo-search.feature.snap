// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searching DuckDuckGo 1`] = `
"[
  [
    { name: 'wedonttrack', value: 'Privacy, simplified.' },
    {
      name: 'content_homepage',
      children: [
        {
          name: 'logo_homepage_link',
          value: '\\\\n\\\\t\\\\t\\\\t\\\\tAbout DuckDuckGo\\\\n\\\\t\\\\t\\\\t\\\\tLearn More\\\\n\\\\t\\\\t\\\\t'
        },
        {
          name: 'search_form_homepage',
          children: [
            { name: 'search_form_input_homepage', value: '' },
            { name: 'search_button_homepage', value: '' },
            { name: 'search_form_input_clear', value: '' },
            { name: 'search_elements_hidden', value: '' }
          ]
        },
        { name: 'error_homepage', value: '' }
      ]
    },
    { name: 'footer_homepage', value: '' },
    {
      name: 'faq',
      children: [
        [
          { name: 'faq-btn-0', value: '' },
          {
            name: 'faq-answer-0',
            value: 'We make our money from private ads on our search engine. On other search engines, ads are based on profiles compiled about you using your personal information like search, browsing, and purchase history. Since we don’t collect that information, search ads on DuckDuckGo are based on the search results page you are viewing, not on you as a person. For example, if you search for cars, we’ll show you ads about cars. Learn more.'
          }
        ],
        [
          { name: 'faq-btn-1', value: '' },
          {
            name: 'faq-answer-1',
            value: 'Not only does Google keep your search history forever by default, their trackers have been found on 75% of the top million websites, which means they are tracking most everywhere you go on the Internet (unless you stop them with DuckDuckGo!). And that’s just the tip of the iceberg. Learn more.'
          }
        ],
        [
          { name: 'faq-btn-2', value: '' },
          {
            name: 'faq-answer-2',
            value: 'Ever notice ads constantly following you around? That’s in part because Google tracks your searches and hides trackers on millions of websites. By contrast, our private search engine doesn’t track your searches and our DuckDuckGo browser extension and mobile app block Google’s (and many other companies’) trackers across the Internet, helping to keep your browsing history more private, as it should be. And that’s just the beginning — by using DuckDuckGo you also escape the manipulation of the filter bubble and can use the Internet faster (after all that tracking code is disabled). Learn more.'
          }
        ],
        [
          { name: 'faq-btn-3', value: '' },
          {
            name: 'faq-answer-3',
            value: 'DuckDuckGo search gives you truly private search results without tradeoffs in result quality. We have everything you’ve come to expect in an online search experience, and a few bonus features that make searching the Internet not only private, but also a bit more fun! And, on top of that, because there is no search history on DuckDuckGo, you escape the filter bubble of manipulated results. Learn more.'
          }
        ],
        [
          { name: 'faq-btn-4', value: '' },
          {
            name: 'faq-answer-4',
            value: 'It is a myth that you can’t be tracked while using so-called “Incognito” mode. In fact, Incognito mode mainly just deletes information on your computer and does nothing to stop Google from saving your searches, nor does it stop companies, Internet service providers, or governments from being able to track you across the Internet. By contrast, DuckDuckGo search is completely anonymous and if you add our app & extension on top, we help keep you private when browsing off of search results. Learn more.'
          }
        ],
        [
          { name: 'faq-btn-5', value: '' },
          {
            name: 'faq-answer-5',
            value: 'We don’t track our users, so it’s actually impossible for us to know how many users in total are using our products. However, we are able to make an estimate based on the number of searches we get per month, which we make public. Our best guess — over 50 million people! Learn more.'
          }
        ],
        [
          { name: 'faq-btn-6', value: '' },
          {
            name: 'faq-answer-6',
            value: 'The DuckDuckGo Privacy Browser mobile app and Privacy Essentials desktop extension both come with our best-in-class tracker blocker, encryption enforcer, and private search engine – all in one simple package. This gives you all the key privacy protection tools to search and browse privately, curbing the constant monitoring of your Internet activity by companies like Google and Facebook. Learn more.'
          }
        ]
      ]
    }
  ],
  { name: 'iframe_hidden', value: '' }
]"
`;

exports[`Searching DuckDuckGo again 1`] = `
"[
  [
    { name: 'wedonttrack', value: 'Privacy, simplified.' },
    {
      name: 'content_homepage',
      children: [
        {
          name: 'logo_homepage_link',
          value: '\\\\n\\\\t\\\\t\\\\t\\\\tAbout DuckDuckGo\\\\n\\\\t\\\\t\\\\t\\\\tLearn More\\\\n\\\\t\\\\t\\\\t'
        },
        {
          name: 'search_form_homepage',
          children: [
            { name: 'search_form_input_homepage', value: '' },
            { name: 'search_button_homepage', value: '' },
            { name: 'search_form_input_clear', value: '' },
            { name: 'search_elements_hidden', value: '' }
          ]
        },
        { name: 'error_homepage', value: '' }
      ]
    },
    { name: 'footer_homepage', value: '' },
    {
      name: 'faq',
      children: [
        [
          { name: 'faq-btn-0', value: '' },
          {
            name: 'faq-answer-0',
            value: 'We make our money from private ads on our search engine. On other search engines, ads are based on profiles compiled about you using your personal information like search, browsing, and purchase history. Since we don’t collect that information, search ads on DuckDuckGo are based on the search results page you are viewing, not on you as a person. For example, if you search for cars, we’ll show you ads about cars. Learn more.'
          }
        ],
        [
          { name: 'faq-btn-1', value: '' },
          {
            name: 'faq-answer-1',
            value: 'Not only does Google keep your search history forever by default, their trackers have been found on 75% of the top million websites, which means they are tracking most everywhere you go on the Internet (unless you stop them with DuckDuckGo!). And that’s just the tip of the iceberg. Learn more.'
          }
        ],
        [
          { name: 'faq-btn-2', value: '' },
          {
            name: 'faq-answer-2',
            value: 'Ever notice ads constantly following you around? That’s in part because Google tracks your searches and hides trackers on millions of websites. By contrast, our private search engine doesn’t track your searches and our DuckDuckGo browser extension and mobile app block Google’s (and many other companies’) trackers across the Internet, helping to keep your browsing history more private, as it should be. And that’s just the beginning — by using DuckDuckGo you also escape the manipulation of the filter bubble and can use the Internet faster (after all that tracking code is disabled). Learn more.'
          }
        ],
        [
          { name: 'faq-btn-3', value: '' },
          {
            name: 'faq-answer-3',
            value: 'DuckDuckGo search gives you truly private search results without tradeoffs in result quality. We have everything you’ve come to expect in an online search experience, and a few bonus features that make searching the Internet not only private, but also a bit more fun! And, on top of that, because there is no search history on DuckDuckGo, you escape the filter bubble of manipulated results. Learn more.'
          }
        ],
        [
          { name: 'faq-btn-4', value: '' },
          {
            name: 'faq-answer-4',
            value: 'It is a myth that you can’t be tracked while using so-called “Incognito” mode. In fact, Incognito mode mainly just deletes information on your computer and does nothing to stop Google from saving your searches, nor does it stop companies, Internet service providers, or governments from being able to track you across the Internet. By contrast, DuckDuckGo search is completely anonymous and if you add our app & extension on top, we help keep you private when browsing off of search results. Learn more.'
          }
        ],
        [
          { name: 'faq-btn-5', value: '' },
          {
            name: 'faq-answer-5',
            value: 'We don’t track our users, so it’s actually impossible for us to know how many users in total are using our products. However, we are able to make an estimate based on the number of searches we get per month, which we make public. Our best guess — over 50 million people! Learn more.'
          }
        ],
        [
          { name: 'faq-btn-6', value: '' },
          {
            name: 'faq-answer-6',
            value: 'The DuckDuckGo Privacy Browser mobile app and Privacy Essentials desktop extension both come with our best-in-class tracker blocker, encryption enforcer, and private search engine – all in one simple package. This gives you all the key privacy protection tools to search and browse privately, curbing the constant monitoring of your Internet activity by companies like Google and Facebook. Learn more.'
          }
        ]
      ]
    }
  ],
  { name: 'iframe_hidden', value: '' }
]"
`;
